test/common/schemaform/FileField.unit.spec.jsx

```diff
@@ -0,0 +1,243 @@
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        uploading: true
+      }
+    ];
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.subTree('.va-growable-background').text()).to.contain('Uploading file...');
+    expect(tree.everySubTree('button')).to.be.empty;
+  });
+
+  it('should render error', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        errorMessage: 'asdfas'
+      }
+    ];
+    const errorSchema = {
+      0: {
+        __errors: [
+          'Bad error'
+        ]
+      }
+    };
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          errorSchema={errorSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.subTree('.va-growable-background').text()).to.contain('Bad error');
+  });
+
+  it('should not render upload button if over max items', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {
+      maxItems: 1
+    };
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        confirmationCode: 'asdfds',
+        name: 'Test file name'
+      }
+    ];
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.everySubTree('label')).to.be.empty;
+  });
+  it('should show edit state', () => {
+    const uiSchema = fileUploadUI('Files');
+    const schema = {
+      type: 'object',
+      properties: {
+        fileField: fileSchema
+      }
+    };
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{
+            fileField: [
+              {
+                confirmationCode: 'asdfasfd'
+              }
+            ]
+          }}
+          uiSchema={{
+            fileField: uiSchema
+          }}/>
+    );
+    const formDOM = getFormDOM(form);
+
+    formDOM.click('button.usa-button-outline');
+
+    expect(formDOM.querySelectorAll('.usa-button').length).to.equal(2);
+    expect(formDOM.querySelectorAll('.usa-button')[0].textContent).to.equal('Replace');
+    expect(formDOM.querySelectorAll('.usa-button')[1].textContent).to.equal('Cancel');
```

Comment:
Nit: Could do a `const buttons = formDOM.querySelectorAll('.usa-button')` and use that in these three `expect`ations.